src/main/java/by/custom/utilcalculator/service/BotFieldsManager.java

```diff
@@ -0,0 +1,100 @@
+package by.custom.utilcalculator.service;
+
+import by.custom.utilcalculator.directory.BotEntity;
+import by.custom.utilcalculator.directory.resources.Commands;
+import by.custom.utilcalculator.directory.steps.*;
+import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
+import org.telegram.telegrambots.meta.api.objects.Message;
+
+public class BotFieldsManager {
+    private final BotEntity botEntity;
+    private final MessagesManager messagesManager;
+
+    public BotFieldsManager() {
+        botEntity = BotEntity.getInstance();
+        messagesManager = MessagesManager.getInstance();
+    }
+
+    public static BotFieldsManager getInstance() {
+        return BotFieldsManagerHolder.BOT_FIELDS_MANAGER;
+    }
+
+    public SendMessage setCarOriginFieldAndGetNextMessageForUser(Message inputMessage) {
```

Comment:
I prefer that BotFieldsManager doesn't know about telegram-specific classes, `SendMessage`, `Message`.
```suggestion
    public String setCarOriginFieldAndGetNextMessageForUser(User sender, command: Commands) {
```

Reply:
I fixed it, but in a little another way. Now BotFieldsManager doesn't know anything about telegram-specific classes. 
Your suggestion is: 
public String setCarOriginFieldAndGetNextMessageForUser**(User sender, command: Commands)** {
My variant is: 
public String setCarOriginFieldAndGetNextMessageForUser(**String command**) {
I am not sure if  I am right. But in my opinion we don't need to add here anything about sender - we can add it in the MessageRouter class. And we don't need to give array: we send only one command. In my opinion my variant is better, because we send less options into the methods. But probably I made a mistake. Please, check it.

Reply:
You are absolutely right. But finally we need user info to save user progress. But it's another story.